tests/functional/lti_certification/v13/conftest.py

```diff
@@ -0,0 +1,173 @@
+            "service_versions": ["2.0"],
+        },
+        "https://purl.imsglobal.org/spec/lti-ags/claim/endpoint": {
+            "scope": [
+                "https://purl.imsglobal.org/spec/lti-ags/scope/lineitem",
+                "https://purl.imsglobal.org/spec/lti-ags/scope/result.readonly",
+                "https://purl.imsglobal.org/spec/lti-ags/scope/score",
+            ],
+            "lineitems": "https://ltiadvantagevalidator.imsglobal.org/ltitool/rest/assignmentsgrades/10843/lineitems",
+        },
+    }
+
+
+@pytest.fixture
+def teacher_payload():
+    now = int(datetime.timestamp(datetime.now()))
+    return {
+        "iss": "https://ltiadvantagevalidator.imsglobal.org",
+        "sub": "TEACHER_ID",
+        "aud": "imstester_4ba76ab",
+        "exp": now + 60,
+        "iat": now,
+        "nonce": "6f889151-c368-4e32-91af-5fe831ee266b",
+        "name": "TEACHER_FIRST_NAME TEACHER_MIDDLE_NAME TEACHER_LAST_NAME",
+        "given_name": "TEACHER_FIRST_NAME",
+        "family_name": "TEACHER_LAST_NAME",
+        "middle_name": "TEACHER_MIDDLE_NAME",
+        "email": "teacher@email.com",
+        "locale": "en-US",
+        "https://purl.imsglobal.org/spec/lti/claim/target_link_uri": "https://localhost/lti_launches",
+        "https://purl.imsglobal.org/spec/lti/claim/deployment_id": "testdeploy",
+        "https://purl.imsglobal.org/spec/lti/claim/message_type": "LtiResourceLinkRequest",
+        "https://purl.imsglobal.org/spec/lti/claim/version": "1.3.0",
+        "https://purl.imsglobal.org/spec/lti/claim/roles": [
+            "http://purl.imsglobal.org/vocab/lis/v2/membership#Instructor"
+        ],
+        "https://purl.imsglobal.org/spec/lti/claim/context": {
+            "id": "COURSE_ID",
+            "label": "COURSE_LABEL",
+            "title": "COURSE_TITLE",
+            "type": ["http://purl.imsglobal.org/vocab/lis/v2/course#CourseSection"],
+        },
+        "https://purl.imsglobal.org/spec/lti/claim/resource_link": {
+            "id": "RESOURCE_ID",
+            "title": "Introduction Assignment",
+            "description": "This is the introduction assignment",
+        },
+        "https://purl.imsglobal.org/spec/lti-nrps/claim/namesroleservice": {
+            "context_memberships_url": "https://ltiadvantagevalidator.imsglobal.org/ltitool/namesandroles.html?memberships=10843",
+            "service_versions": ["2.0"],
+        },
+        "https://purl.imsglobal.org/spec/lti-ags/claim/endpoint": {
+            "scope": [
+                "https://purl.imsglobal.org/spec/lti-ags/scope/lineitem",
+                "https://purl.imsglobal.org/spec/lti-ags/scope/result.readonly",
+                "https://purl.imsglobal.org/spec/lti-ags/scope/score",
+            ],
+            "lineitems": "https://ltiadvantagevalidator.imsglobal.org/ltitool/rest/assignmentsgrades/10843/lineitems",
+        },
+    }
```

Comment:
```suggestion
@pytest.fixture
def student_payload(common_payload):
    return {
        "sub": "STUDENT_ID",
        "nonce": "d45a2497-4389-45ea-8d61-b1fa7e407447",
        "name": "STUDENT_FIRST_NAME STUDENT_MIDDLE_NAME STUDENT_LAST_NAME",
        "given_name": "STUDENT_FIRST_NAME",
        "family_name": "STUDENT_LAST_NAME",
        "middle_name": "STUDENT_MIDDLE_NAME",
        "email": "student@email.com",
        "https://purl.imsglobal.org/spec/lti/claim/roles": [
            "http://purl.imsglobal.org/vocab/lis/v2/membership#Learner"
        ],
        **common_payload,
    }


@pytest.fixture
def teacher_payload(common_payload):
    return {
        "sub": "TEACHER_ID",
        "nonce": "6f889151-c368-4e32-91af-5fe831ee266b",
        "name": "TEACHER_FIRST_NAME TEACHER_MIDDLE_NAME TEACHER_LAST_NAME",
        "given_name": "TEACHER_FIRST_NAME",
        "family_name": "TEACHER_LAST_NAME",
        "middle_name": "TEACHER_MIDDLE_NAME",
        "email": "teacher@email.com",
        "https://purl.imsglobal.org/spec/lti/claim/roles": [
            "http://purl.imsglobal.org/vocab/lis/v2/membership#Instructor"
        ],
        **common_payload,
    }


@pytest.fixture
def common_payload():
    # This is the parts of the payload common to both teacher and student
    now = int(datetime.timestamp(datetime.now()))

    return {
        "exp": now + 60,
        "iat": now,
        "iss": "https://ltiadvantagevalidator.imsglobal.org",
        "aud": "imstester_4ba76ab",
        "locale": "en-US",
        "https://purl.imsglobal.org/spec/lti/claim/target_link_uri": "https://localhost/lti_launches",
        "https://purl.imsglobal.org/spec/lti/claim/deployment_id": "testdeploy",
        "https://purl.imsglobal.org/spec/lti/claim/message_type": "LtiResourceLinkRequest",
        "https://purl.imsglobal.org/spec/lti/claim/version": "1.3.0",
        "https://purl.imsglobal.org/spec/lti/claim/context": {
            "id": "COURSE_ID",
            "label": "COURSE_LABEL",
            "title": "COURSE_TITLE",
            "type": ["http://purl.imsglobal.org/vocab/lis/v2/course#CourseSection"],
        },
        "https://purl.imsglobal.org/spec/lti/claim/resource_link": {
            "id": "RESOURCE_ID",
            "title": "Introduction Assignment",
            "description": "This is the introduction assignment",
        },
        "https://purl.imsglobal.org/spec/lti-nrps/claim/namesroleservice": {
            "context_memberships_url": "https://ltiadvantagevalidator.imsglobal.org/ltitool/namesandroles.html?memberships=10843",
            "service_versions": ["2.0"],
        },
        "https://purl.imsglobal.org/spec/lti-ags/claim/endpoint": {
            "scope": [
                "https://purl.imsglobal.org/spec/lti-ags/scope/lineitem",
                "https://purl.imsglobal.org/spec/lti-ags/scope/result.readonly",
                "https://purl.imsglobal.org/spec/lti-ags/scope/score",
            ],
            "lineitems": "https://ltiadvantagevalidator.imsglobal.org/ltitool/rest/assignmentsgrades/10843/lineitems",
        },
    }
```

Kind of giant suggestion, but nearly everything about these two fixtures is the same. Not only does this mean we have a lot of duplication, but it's also hard to spot what is different about them or not.